Framework/Algorithms/src/DeleteWorkspaces.cpp

```diff
@@ -35,11 +35,27 @@ void DeleteWorkspaces::exec() {
       // properties were set. If a workspace is missing, it was probably
       // a group workspace whose contents were deleted before the group
       // itself.
-      auto deleteAlg = createChildAlgorithm("DeleteWorkspace", -1, -1, false);
-      deleteAlg->initialize();
-      deleteAlg->setPropertyValue("Workspace", wsName);
-      bool success = deleteAlg->execute();
-      if (!deleteAlg->isExecuted() || !success) {
+      bool success = false;
+      bool executed = false;
+      try {
```

Comment:
Do we know why `DeleteWorkspace` can't just handle this?

Reply:
There's a call to a very long method in `Algorithm`, `checkGroups()`, which thows the exception. It's the running of the `DeleteWorkspace` child algorithm that causes the throw. It could be overridden in `DeleteWorkspace` but it seemed that would be too much of a change for the release at the time